src/graph_impl/mod.rs

```diff
@@ -2010,20 +2011,21 @@ impl<Ix: IndexType> WalkNeighbors<Ix> {
         &mut self,
         g: &Graph<N, E, Ty, Ix>,
     ) -> Option<(EdgeIndex<Ix>, NodeIndex<Ix>)> {
+        let edge_present: for<'r> fn(&'r &Edge<E, Ix>) -> bool =
```

Comment:
This check should be in StableGraph, as before. Graph automatically compresses the nodes on deletion, so there shouldn't be empty ones. (`NodeIndex::end()` may be a valid identifier).

The comment addition should be the only thing needed for `Graph`.

Reply:
Sorry, I didn't see at this should be moved to `stable_graph::WalkNeighbors.next` -- that just calls this function with a one-liner to `self.inner.next`. Would you prefer to change that instead? As far as `Graph` goes, if `NodeIndex::end()` is present for both nodes then it seems like there wouldn't be any adverse consequences for detecting that here, since it would imply no edges remain anyway. Have I missed something?